### PR TITLE
add jest config files

### PR DESCRIPTION
**Pull request by AntonGluschuk**
develop

## GitHub Board

**Issue link**

[ ]( )

**Story link**

[#11 Story]( )

## Code reviewers
- [ ] @github_username

## Summary of issue

- [ ]  Summary of issue
- [ ]  Summary of issue

## Summary of change

- [ ]  Summary of change
- [ ]  Summary of change

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test",
+    "test": "jest --coverage",
```
Comment (SergeOlabin) [inline package.json:32]: we need a separate script for the coverage, can't touch this one it's used for testing and on the CI as well.

File: jest-config/file-mock.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
```
Comment (SergeOlabin) [inline jest-config/file-mock.js:1]: what is this file about?

File: jest-config/style-mock.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
```
Comment (SergeOlabin) [inline jest-config/style-mock.js:1]: what is this file about?

**Review by SergeOlabin — CHANGES_REQUESTED** [review]

File: jest-config/file-mock.js (hunk 2 above)
Comment (AntonGluschuk) [reply jest-config/file-mock.js:1]: required for successful tests using jest script (when some components import external modules e.g. `react-slick`)

File: jest-config/style-mock.js (hunk 3 above)
Comment (AntonGluschuk) [reply jest-config/style-mock.js:1]: required for successful tests using jest script (when some components import external style files with `.css`)

File: package.json (hunk 1 above)
Comment (AntonGluschuk) [reply package.json]: fixed

File: jest.config.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  testMatch: [
+    '**/__tests__/**/*.+(ts|tsx|js)',
+    '**/?(*.)+(spec|test).+(ts|tsx|js)',
+  ],
+  transform: {
+    '^.+\\.(ts|tsx)$': 'ts-jest',
+    '.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
+      '<rootDir>/jest-config/file-mock.js',
+    '.(css|less)$': '<rootDir>/jest-config/style-mock.js',
+  },
```
Comment (calster00) [inline jest.config.js:12]: I suggest we use `identity-obj-proxy `for mocking css imports, as [described](https://jestjs.io/docs/en/webpack#mocking-css-modules) by jest.
```suggestion
  },
    moduleNameMapper: {
    '\\.(css|less)$': 'identity-obj-proxy',
  },
```

**Review by calster00 — CHANGES_REQUESTED** [review]

File: jest.config.js (hunk 4 above)
Comment (AntonGluschuk) [reply jest.config.js]: done

**Pull request closed without merging**